codegen/templates/tchannel_endpoint.tmpl

```diff
@@ -80,7 +86,25 @@ func (h *{{$handlerName}}) Handle(
 	ctx context.Context,
 	reqHeaders map[string]string,
 	wireValue *wire.Value,
-) (bool, zanzibar.RWTStruct, map[string]string, error) {
+) (isSuccessful bool, response zanzibar.RWTStruct, headers map[string]string, e error) {
+	defer func() {
+		if r := recover(); r != nil {
+			stacktrace := string(debug.Stack())
+			e = errors.Errorf("enpoint panic: %v, stacktrace: %v", r, stacktrace)
+			h.Deps.Default.ContextLogger.Error(
+				ctx,
+				"endpoint panic",
+				zap.Error(e),
+				zap.String("stacktrace", stacktrace),
+				zap.String("endpoint", h.endpoint.EndpointID))
+
+			h.endpointScope.Counter("endpoint.panic").Inc(1)
+			isSuccessful = false
+			response = nil
+			headers = map[string]string{}
```

Comment:
nit: do we want to preserve some header information if any

Reply:
return nil instead of empty, this is the same behavior https://github.com/uber/zanzibar/pull/458/files/ef542b0b5925a2d569a7fc6d424379c6d39ff036#diff-7c8f7ec39bf077182fa0b404b3d49d76R124 when other error happens